### ISSUE TITLE
pyvips error 0x7e

### ISSUE THREAD

**Issue by ipr999** [libvips/pyvips#210]
Hello Sir,

I encountered an error in python after running pyvips, I've search the internet for solutions including changing the 'path' in advance system settings and changing the path in the python code, but nothing worked. Here is the error message:

```
Traceback (most recent call last):
  File "C:\Users\Computer\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\LocalCache\local-packages\Python38\site-packages\pyvips\__init__.py", line 19, in <module>
    import _libvips
ModuleNotFoundError: No module named '_libvips'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Computer\Desktop\Cervical Biopsies Cancer Detection Proj\main.py", line 5, in <module>
    import pyvips
  File "C:\Users\Computer\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\LocalCache\local-packages\Python38\site-packages\pyvips\__init__.py", line 70, in <module>
    gobject_lib = ffi.dlopen(_gobject_libname)
  File "C:\Users\Computer\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\LocalCache\local-packages\Python38\site-packages\cffi\api.py", line 150, in dlopen
    lib, function_cache = _make_ffi_library(self, name, flags)
  File "C:\Users\Computer\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\LocalCache\local-packages\Python38\site-packages\cffi\api.py", line 832, in _make_ffi_library
    backendlib = _load_backend_lib(backend, libname, flags)
  File "C:\Users\Computer\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\LocalCache\local-packages\Python38\site-packages\cffi\api.py", line 828, in _load_backend_lib
    return backend.load_library(path, flags)
OSError: cannot load library 'C:\vips-dev-8.10\bin\libgobject-2.0-0.dll': error 0x7e
```

I'm using windows 10 pro and Python 3.8 (64 bit) and libvips 8.10; I installed pyvips using pip.

Your response will be helpful in the code I'm doing. Thank you very much in advance.

Best regards.

**Comment by jcupitt** [comment]
Hello, you need to give more information. How did you set PATH? What program did you try to run?

For example, this program:

```python
import os
vipshome = 'c:\\vips-dev-8.10\\bin'
os.environ['PATH'] = vipshome + ';' + os.environ['PATH']
import pyvips
```

Ought to work (I think).

**Comment by jcupitt** [comment]
Install notes here:

https://libvips.github.io/pyvips/README.html#non-conda-install

**Comment by ipr999** [comment]
![image](https://user-images.githubusercontent.com/53067636/95585971-8e025880-0a72-11eb-9ab6-7f3cd7f6c15a.png)
![image](https://user-images.githubusercontent.com/53067636/95586171-dfaae300-0a72-11eb-9aad-f0cc58917a55.png)

I did all the instruction in https://libvips.github.io/pyvips/README.html#non-conda-install
but nothing really worked

**Comment by jcupitt** [comment]
I would not change your PATH in the environment variable dialog -- it's likely to break other packages because of DLL conflicts.

**Comment by jcupitt** [comment]
Where did you download Python from?

**Comment by ipr999** [comment]
I use cmd, pip install pyvips

**Comment by ipr999** [comment]
![image](https://user-images.githubusercontent.com/53067636/95587685-f5210c80-0a74-11eb-8056-39e152039250.png)

**Comment by jcupitt** [comment]
But where did you get the python language from? From anaconda, or from python.org?

**Comment by ipr999** [comment]
from python.org downloaded it from windows 10 store

**Comment by ipr999** [comment]
![image](https://user-images.githubusercontent.com/53067636/95588035-77113580-0a75-11eb-8418-93bb645305b1.png)

**Comment by jcupitt** [comment]
I've never tried the store version, I tested with the download from python.org, eg.:

https://www.python.org/downloads/release/python-390/

I'll try to find a Windows machine over the weekend and try with the current version.

**Comment by ipr999** [comment]
Okay I'll do the same, I will try the web version from python.org, I will give you feedback once I've installed the software. Thanks.

**Comment by ipr999** [comment]
Hello, I've uninstalled python 3.8 from windows store and replace it with 3.9 from python.org website. The issue is now resolve, pyvips is now working. Thanks for your time. I also took your advice not to change the path in the variables, cheers!

**Comment by jcupitt** [comment]
\o/ that's great. I wonder why the store one is not working?

**Comment by kleisauke** [comment]
I can confirm that pyvips does not work when installing Python from the Microsoft Store. 

This is probably due to the [UWP restrictions](https://docs.python.org/3/using/windows.html#known-issues). This can be resolved by cross-compiling libvips for Windows targeting UWP (i.e. by using [these flags](https://github.com/mstorsjo/llvm-mingw/blob/5fc9728420a1ea45fc2be634a38a3392663790ee/wrappers/clang-target-wrapper.sh#L67-L83)). However, the Windows Store API is very restrictive, which could affect the performance of libvips, see for example these merge requests:
https://gitlab.gnome.org/GNOME/glib/-/merge_requests/951
https://gitlab.gnome.org/GNOME/glib/-/merge_requests/1060
https://gitlab.freedesktop.org/gstreamer/orc/-/merge_requests/42

TL;DR I recommend installing Python from https://www.python.org/.

**Comment by jcupitt** [comment]
Ah I didn't realize it ran (at least partially) in the UWP sandbox. I thought the store allowed plain win32 programs as well now?

Thanks for investigating Kleis.

**Comment by ipr999** [comment]
I have no idea why store version of python is not working with pyvips, I'm a newbie in programming... But thanks for your help, it's now working. 

P.S. I've installed Python 3.9 from python.org website but I've downgraded to 3.8 due to issues on pip installation of some modules, but as I run 3.8 version of Python, pyvips is still working.
